src/libraries/System.Net.Http.Json/ref/System.Net.Http.Json.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// ------------------------------------------------------------------------------
+// Changes to this file must follow the https://aka.ms/api-review process.
+// ------------------------------------------------------------------------------
+
+namespace System.Net.Http.Json
+{
+    public static partial class HttpClientJsonExtensions
+    {
+        public static System.Threading.Tasks.Task<object?> GetFromJsonAsync(this System.Net.Http.HttpClient client, string requestUri, System.Type type, System.Text.Json.JsonSerializerOptions? options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<object?> GetFromJsonAsync(this System.Net.Http.HttpClient client, System.Uri requestUri, System.Type type, System.Text.Json.JsonSerializerOptions? options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<T> GetFromJsonAsync<T>(this System.Net.Http.HttpClient client, string requestUri, System.Text.Json.JsonSerializerOptions? options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<T> GetFromJsonAsync<T>(this System.Net.Http.HttpClient client, System.Uri requestUri, System.Text.Json.JsonSerializerOptions? options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
Interesting design question for @terrajobst and @stephentoub - do have a clear picture on what the right thing is to do with `T` regarding nullability? 

Suppose that I call `GetFromJsonAsync<Widget>(...)` where `Widget` is a class. The JSON response could just be the null-literal `null` which would deserialize to a null-valued `Widget`. 

AFAIK there's no to reflect the fact that this API fundamentally can always return a null value without constraining `T` to reference types only right?. I assume the serializer APIs have the same issue/concern.

Reply:
We don't have any way to express that.  For the sync methods, we can attribute the result T as [MaybeNull], but for the async methods, we lack the expressivity to do so when the T is a generic type parameter.  This might end up being addressed in C# 9 if ?? sees the light of day.
cc: @jcouv